crates/mock-download-server/src/handlers.rs

```diff
@@ -1,164 +1,106 @@
 // SPDX-FileCopyrightText: The Ferrocene Developers
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use crate::Serialize;
+use std::str::FromStr;
+use std::sync::Arc;
+
 use crate::{AuthenticationToken, Data};
+use axum::extract::{Path, State};
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::Json;
+use axum_extra::headers::authorization::Bearer;
+use axum_extra::headers::{self, IfNoneMatch};
+use axum_extra::{headers::Authorization, TypedHeader};
 use criticaltrust::manifests::ManifestVersion;
 use md5::Digest;
-use tiny_http::{Header, Method, Request, Response, ResponseBox, StatusCode};
+use tokio::sync::Mutex;
+
+pub(crate) async fn handle_v1_package(
+    State(data): State<Arc<Mutex<Data>>>,
+    if_none_match: Option<TypedHeader<IfNoneMatch>>,
+    Path((product, release, package, _format)): Path<(String, String, String, String)>,
+) -> impl IntoResponse {
+    let data = data.lock().await;
+
+    let bytes = data
+        .release_packages
+        .get(&(
+            product.to_string(),
+            release.to_string(),
+            package.to_string(),
+        ))
+        .unwrap()
+        .clone();
+
+    if let Some(if_none_match) = if_none_match {
```

Comment:
@emig I added the etag code here instead of keys since that is where we primarily use it in 'real'.